### ISSUE TITLE
Arcing trajectories or very slow missiles break inherit parent target.

### ISSUE THREAD

**Issue by EatThePath** [scp-fs2open/fs2open.github.com#3666]
This is context for #3661, which fixes it, but I fiugred it'd be good to have why documented somewhere besides discord.

I have a weapon in Warmachine that I added as VLS to a Ravana. I've had this missile on other ships and it's worked fine, but in this case it breaks.
https://cdn.discordapp.com/attachments/223511391531827200/888276744585494558/screen0490.png
The weapon is a staged weapon, spawning a cluster of homing submunition. In this pictured case both streams home on their target fine, but depending on how I tune the situation one or both of these streams of missiles will spawn it's submunitions without any targeting.
I debugged this, and at spawn time the target_num for the spawner weapon is -1 for the duds, so that's the result I would expect. It still homes though.

After digging I find this lovely code in weapons.cpp, currently line 5458, in weapons_process_post.
```cpp
if (cur_dist < wp->nearest_dist) {
    wp->nearest_dist = cur_dist;
} else if (cur_dist > wp->nearest_dist + 1.0f) {
    float        dot;
    vec3d    tvec;
    ai_info    *parent_aip;
    parent_aip = NULL;
    if (obj->parent != Player_obj-Objects) {
        parent_aip = &Ai_info[Ships[Objects[obj->parent].instance].ai_index];
    }
    vm_vec_normalized_dir(&tvec, &v0, &Objects[wp->target_num].pos);
    dot = vm_vec_dot(&tvec, &Objects[wp->target_num].orient.vec.fvec);
    wp->target_num = -1;
```

This code gets tripped during the flight. It is apparently concerned with having the AI adjust their aim when a missile misses, but there's no evident good reason for clearing the target num. Homing is tied to target signature first of all, and that's not cleared, so the missile keeps homing, but spawning passes target by object num alone.

Because of the wide arc, the sharp angle of the launch tubes, and the long accelerations/turnings I have, the target is sometimes pulling away from the missile before the missile starts to catch up. If I comment that out, everything starts working again.

**Comment by z64555** [comment]
Ok, current plan is to try to tackle this as a point release after 21.4 is completed

**Comment by Goober5000** [comment]
Fixed in #3661.
